prinz/src/main/scala/pl/touk/nussknacker/prinz/enrichers/PrinzEnricher.scala

```diff
@@ -1,19 +1,29 @@
 package pl.touk.nussknacker.prinz.enrichers
 
-import javax.validation.constraints.NotBlank
-import pl.touk.nussknacker.engine.api.{MetaData, MethodToInvoke, ParamName, Service}
+import pl.touk.nussknacker.engine.api.definition.{Parameter, ServiceWithExplicitMethod}
+import pl.touk.nussknacker.engine.api.test.InvocationCollectors
+import pl.touk.nussknacker.engine.api.typed.typing
+import pl.touk.nussknacker.engine.api.typed.typing.Typed
+import pl.touk.nussknacker.engine.api.{ContextId, MetaData}
+import pl.touk.nussknacker.prinz.model.Model
 
 import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
 
-class PrinzEnricher extends Service {
+class PrinzEnricher(private val model: Model) extends ServiceWithExplicitMethod {
 
-  @MethodToInvoke
-  def invoke(@ParamName("sepalLength") @NotBlank sepalLength: Float,
-             @ParamName("sepalWidth") @NotBlank sepalWidth: Float,
-             @ParamName("petalLength") @NotBlank petalLength: Float,
-             @ParamName("petalWidth") @NotBlank petalWidth: Float)
-            (implicit ec: ExecutionContext, metaData: MetaData): Future[String] = {
-    Future { "setosa" }
+  override def invokeService(params: List[AnyRef])
+                            (implicit ec: ExecutionContext, collector: InvocationCollectors.ServiceInvocationCollector,
+                             metaData: MetaData, contextId: ContextId): Future[AnyRef] = {
+    val res = ().asInstanceOf[AnyRef]
+    Future.successful(res)
   }
+
+  override def parameterDefinition: List[Parameter] =
+    List(
+      Parameter[Float]("inputA"),
```

Comment:
define based on model signature

Reply:
for now Enricher has some dummy code, it was not part of this task. i think it should be done after changing model run result to return Future and also when we'll be able to deploy nussknacker process and test out enricher

Reply:
ok, understood